### ISSUE TITLE
TST: test failure and noise in statsmodels.stats

### ISSUE THREAD

**Issue by josef-pkt** [statsmodels/statsmodels#6796]
** update ** unit test in `test_moment_helpers.py` doesn't set seed correctly, see comments below

I get this on my machine with #6789 recently rebased on master

I didn't look at the details yet.
main problems invalid value in sqrt and zero division

```
================================== FAILURES ===================================
______________________ test_multidimensional[test_vals1] ______________________

test_vals = array([[79, 37, 13],
       [ 0, 75, 79],
       [91, 86, 81],
       [50, 64, 57]])

    @pytest.mark.parametrize('test_vals', multidimension_test_vals)
    def test_multidimensional(test_vals):
        assert_almost_equal(cum2mc(mnc2cum(mc2mnc(test_vals).T).T).T, test_vals

        assert_almost_equal(cum2mc(mc2cum(test_vals).T).T, test_vals)
>       assert_almost_equal(mvsk2mc(mc2mvsk(test_vals).T).T, test_vals)

..\statsmodels\statsmodels_py37\statsmodels\statsmodels\stats\tests\
est_moment_helpers.py:105:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

x = array([[79., 37., 13.],
       [ 0., 75., 79.],
       [nan, 86., 81.],
       [nan, 64., 57.]])
y = array([[79, 37, 13],
       [ 0, 75, 79],
       [91, 86, 81],
       [50, 64, 57]])
func = <ufunc 'isnan'>, hasval = 'nan'

    def func_assert_same_pos(x, y, func=isnan, hasval='nan'):
        """Handling nan/inf.

        Combine results of running func on x and y, checking that they are True
        at the same locations.

        """
        x_id = func(x)
        y_id = func(y)
        # We include work-arounds here to handle three types of slightly
        # pathological ndarray subclasses:
        # (1) all() on `masked` array scalars can return masked arrays, so we
        #     use != True
        # (2) __eq__ on some ndarray subclasses returns Python booleans
        #     instead of element-wise comparisons, so we cast to bool_() and
        #     use isinstance(..., bool) checks
        # (3) subclasses with bare-bones __array_function__ implemenations may
        #     not implement np.all(), so favor using the .all() method
        # We are not committed to supporting such subclasses, but it's nice to
        # support them if possible.
        if bool_(x_id == y_id).all() != True:
            msg = build_err_msg([x, y],
                                err_msg + '\nx and y %s location mismatch:'
                                % (hasval), verbose=verbose, header=header,
                                names=('x', 'y'), precision=precision)
>           raise AssertionError(msg)
E           AssertionError:
E           Arrays are not almost equal to 7 decimals
E
E           x and y nan location mismatch:
E            x: array([[79., 37., 13.],
E                  [ 0., 75., 79.],
E                  [nan, 86., 81.],
E                  [nan, 64., 57.]])
E            y: array([[79, 37, 13],
E                  [ 0, 75, 79],
E                  [91, 86, 81],
E                  [50, 64, 57]])

...\python-3.7.4.amd64\lib\site-packages\numpy\
esting\_private\utils.py:735: AssertionError
============================== warnings summary ===============================
statsmodels/stats/tests/test_anova.py::TestAnovaLMCompare::test_results
statsmodels/stats/tests/test_anova.py::TestAnovaLMCompareNoconstant::test_resul
s
statsmodels/stats/tests/test_contingency_tables.py::TestStratified1::test_panda

statsmodels/stats/tests/test_diagnostic.py::TestDiagnosticG::test_harvey_collie

statsmodels/stats/tests/test_diagnostic.py::TestDiagnosticGPandas::test_harvey_
ollier
statsmodels/stats/tests/test_power.py::TestFtestAnovaPower::test_power_plot
statsmodels/stats/tests/test_power.py::test_power_solver_warn
statsmodels/stats/tests/test_proportion.py::test_confint_2indep
statsmodels/stats/tests/test_proportion.py::test_confint_2indep_propcis
statsmodels/stats/tests/test_proportion.py::test_test_2indep
statsmodels/stats/tests/test_proportion.py::test_equivalence_2indep
statsmodels/stats/tests/test_tost.py::test_tost_asym
  ...\python-3.7.4.amd64\lib\site-packages\scip
\stats\_distn_infrastructure.py:901: RuntimeWarning: invalid value encountered
n greater
    return (a < x) & (x < b)

statsmodels/stats/tests/test_anova.py::TestAnovaLMCompare::test_results
statsmodels/stats/tests/test_anova.py::TestAnovaLMCompareNoconstant::test_resul
s
statsmodels/stats/tests/test_contingency_tables.py::TestStratified1::test_panda

statsmodels/stats/tests/test_diagnostic.py::TestDiagnosticG::test_harvey_collie

statsmodels/stats/tests/test_diagnostic.py::TestDiagnosticGPandas::test_harvey_
ollier
statsmodels/stats/tests/test_power.py::TestFtestAnovaPower::test_power_plot
statsmodels/stats/tests/test_power.py::test_power_solver_warn
statsmodels/stats/tests/test_proportion.py::test_confint_2indep
statsmodels/stats/tests/test_proportion.py::test_confint_2indep_propcis
statsmodels/stats/tests/test_proportion.py::test_test_2indep
statsmodels/stats/tests/test_proportion.py::test_equivalence_2indep
statsmodels/stats/tests/test_tost.py::test_tost_asym
  ...\python-3.7.4.amd64\lib\site-packages\scip
\stats\_distn_infrastructure.py:901: RuntimeWarning: invalid value encountered
n less
    return (a < x) & (x < b)

statsmodels/stats/tests/test_anova.py::TestAnovaLMCompare::test_results
statsmodels/stats/tests/test_anova.py::TestAnovaLMCompareNoconstant::test_resul
s
statsmodels/stats/tests/test_diagnostic.py::TestDiagnosticG::test_harvey_collie

statsmodels/stats/tests/test_diagnostic.py::TestDiagnosticGPandas::test_harvey_
ollier
statsmodels/stats/tests/test_power.py::TestFtestAnovaPower::test_power_plot
statsmodels/stats/tests/test_power.py::test_power_solver_warn
statsmodels/stats/tests/test_proportion.py::test_confint_2indep
statsmodels/stats/tests/test_proportion.py::test_confint_2indep_propcis
statsmodels/stats/tests/test_proportion.py::test_test_2indep
statsmodels/stats/tests/test_proportion.py::test_equivalence_2indep
statsmodels/stats/tests/test_tost.py::test_tost_asym
  ...\python-3.7.4.amd64\lib\site-packages\scip
\stats\_distn_infrastructure.py:1892: RuntimeWarning: invalid value encountered
in less_equal
    cond2 = cond0 & (x <= _a)

statsmodels/stats/tests/test_contingency_tables.py::TestStratified1::test_panda

  ...\statsmodels\statsmodels_py37\statsmodels\statsmodels\
tats\contingency_tables.py:1216: RuntimeWarning: divide by zero encountered in
rue_divide
    v11 = (1 / e11 + 1 / (self._apc - e11) + 1 / (self._apb - e11) +

statsmodels/stats/tests/test_contingency_tables.py::TestStratified1::test_panda

  ...\statsmodels\statsmodels_py37\statsmodels\statsmodels\
tats\contingency_tables.py:1216: RuntimeWarning: invalid value encountered in a
d
    v11 = (1 / e11 + 1 / (self._apc - e11) + 1 / (self._apb - e11) +

statsmodels/stats/tests/test_contingency_tables.py::TestStratified1::test_panda

  ...\statsmodels\statsmodels_py37\statsmodels\statsmodels\
tats\contingency_tables.py:1217: RuntimeWarning: divide by zero encountered in
rue_divide
    1 / (self._dma + e11))

statsmodels/stats/tests/test_contingency_tables.py::TestStratified1::test_panda

  ...\statsmodels\statsmodels_py37\statsmodels\statsmodels\
tats\contingency_tables.py:1220: RuntimeWarning: invalid value encountered in t
ue_divide
    statistic = np.sum((table[0, 0, :] - e11)**2 / v11)

statsmodels/stats/tests/test_contingency_tables.py::TestStratified1::test_panda

  ...\python-3.7.4.amd64\lib\site-packages\scip
\stats\_distn_infrastructure.py:1807: RuntimeWarning: invalid value encountered
in greater_equal
    cond2 = (x >= _b) & cond0

statsmodels/stats/tests/test_corrpsd.py::TestCovPSD::test_cov_nearest
statsmodels/stats/tests/test_corrpsd.py::TestCorrPSD1::test_nearest
statsmodels/stats/tests/test_corrpsd.py::test_corrpsd_threshold[0]
statsmodels/stats/tests/test_corrpsd.py::test_corrpsd_threshold[1e-15]
statsmodels/stats/tests/test_corrpsd.py::test_corrpsd_threshold[1e-10]
statsmodels/stats/tests/test_corrpsd.py::test_corrpsd_threshold[1e-06]
  ...\statsmodels\statsmodels_py37\statsmodels\statsmodels\
tats\correlation_tools.py:90: IterationLimitWarning:
  Maximum iteration reached.

    warnings.warn(iteration_limit_doc, IterationLimitWarning)

statsmodels/stats/tests/test_diagnostic.py::TestDiagnosticG::test_normality
statsmodels/stats/tests/test_diagnostic.py::TestDiagnosticGPandas::test_normali
y
  ...\statsmodels\statsmodels_py37\statsmodels\statsmodels\
tats\_adnorm.py:70: RuntimeWarning: divide by zero encountered in log1p
    s = np.sum((2 * i[sl1] - 1.0) / nobs * (np.log(z) + np.log1p(-z[sl2])),

statsmodels/stats/tests/test_dist_dependant_measures.py::TestDistDependenceMeas
res::test_statistic_value_emp_method
  ...s\statsmodels\statsmodels_py37\statsmodels\statsmodels\
tats\dist_dependence_measures.py:137: UserWarning: p-value was 0.0 when using t
e empirical method. The asymptotic approximation will be used instead
    warnings.warn(msg)

statsmodels/stats/tests/test_moment_helpers.py::test_multidimensional[test_vals
]
  ...\statsmodels\statsmodels_py37\statsmodels\statsmodels\
tats\moment_helpers.py:188: RuntimeWarning: divide by zero encountered in true_
ivide
    skew = np.divide(mc3, mc2 ** 1.5)

statsmodels/stats/tests/test_moment_helpers.py::test_multidimensional[test_vals
]
  ...\statsmodels\statsmodels_py37\statsmodels\statsmodels\
tats\moment_helpers.py:189: RuntimeWarning: divide by zero encountered in true_
ivide
    kurt = np.divide(mc4, mc2 ** 2.0) - 3.0

statsmodels/stats/tests/test_moment_helpers.py::test_multidimensional[test_vals
]
  ...\statsmodels\statsmodels_py37\statsmodels\statsmodels\
tats\moment_helpers.py:154: RuntimeWarning: invalid value encountered in double
scalars
    cnt[2] = sk * sig2 ** 1.5

statsmodels/stats/tests/test_moment_helpers.py::test_multidimensional[test_vals
]
  ...\statsmodels\statsmodels_py37\statsmodels\statsmodels\
tats\moment_helpers.py:155: RuntimeWarning: invalid value encountered in double
scalars
    cnt[3] = (kur + 3.0) * sig2 ** 2.0

statsmodels/stats/tests/test_power.py::test_power_solver_warn
  ...\statsmodels\statsmodels_py37\statsmodels\statsmodels\
tats\power.py:107: RuntimeWarning: invalid value encountered in sqrt
    pow_ = stats.norm.sf(crit - d*np.sqrt(nobs)/sigma)

statsmodels/stats/tests/test_proportion.py::test_confint_2indep
statsmodels/stats/tests/test_proportion.py::test_confint_2indep_propcis
statsmodels/stats/tests/test_proportion.py::test_test_2indep
statsmodels/stats/tests/test_proportion.py::test_equivalence_2indep
  ...\statsmodels\statsmodels_py37\statsmodels\statsmodels\
tats\proportion.py:1388: RuntimeWarning: divide by zero encountered in double_s
alars
    var = (1 / (prop1 * (1 - prop1) * nobs1) +

statsmodels/stats/tests/test_proportion.py::test_confint_2indep
statsmodels/stats/tests/test_proportion.py::test_confint_2indep_propcis
statsmodels/stats/tests/test_proportion.py::test_test_2indep
statsmodels/stats/tests/test_proportion.py::test_equivalence_2indep
  ...\statsmodels\statsmodels_py37\statsmodels\statsmodels\
tats\proportion.py:1393: RuntimeWarning: divide by zero encountered in double_s
alars
    diff_stat = ((p1 - prop1) / (prop1 * (1 - prop1)) -

statsmodels/stats/tests/test_proportion.py::test_confint_2indep
statsmodels/stats/tests/test_proportion.py::test_confint_2indep_propcis
statsmodels/stats/tests/test_proportion.py::test_test_2indep
statsmodels/stats/tests/test_proportion.py::test_equivalence_2indep
  ...\statsmodels\statsmodels_py37\statsmodels\statsmodels\
tats\weightstats.py:671: RuntimeWarning: invalid value encountered in double_sc
lars
    zstat = value / std_diff

statsmodels/stats/tests/test_tost.py::test_tost_asym
  ...\statsmodels\statsmodels_py37\statsmodels\statsmodels\
tats\weightstats.py:1298: RuntimeWarning: invalid value encountered in log
    low = transform(low)

-- Docs: https://docs.pytest.org/en/latest/warnings.html
= 1 failed, 1117 passed, 1 skipped, 2 xfailed, 67 warnings in 100.64 seconds ==
```

**Comment by josef-pkt** [comment]
I didn't get anymore the failure in `test_multidimensional` in two more runs of those unit tests.
I cannot replicate the test numbers.

There is a silly bug in the unit tests that doesn't set the seed for the test case.
The code defines a RandomState with seed but doesn't use it, i.e. uses global random state for sampling.

https://github.com/statsmodels/statsmodels/blob/master/statsmodels/stats/tests/test_moment_helpers.py#L92

```
rs = np.random.RandomState(12345)
random_vals = np.random.randint(0, 100, 12).reshape(4, 3)
```

**Comment by bashtage** [comment]
Closed by #6835
